src/bellmanford.jl

```diff
@@ -105,3 +108,28 @@ function has_negative_edge_cycle{V, D}(
     edge_inspector = VectorEdgePropertyInspector{D}(edge_dists)
     has_negative_edge_cycle(graph, edge_inspector)
 end
+
+function enumerate_paths{V,D<:Number}(state::BellmanFordStates{V,D}, dest::Vector{V})
+    parents = state.parents
+    hasparent = state.hasparent
+    
+    num_dest = length(dest)
+    all_paths = Array(Vector{V},num_dest)
+    for i=1:num_dest
+        all_paths[i] = V[]
+        index = dest[i]
+        if hasparent[index] || parents[index] == index
+            while hasparent[index]
```

Comment:
I was going to use it for betweenness centrality, but both my _explicit and @yeesian's suffers from two problems:

1) multiple shortest paths are ignored; and
2) there are more efficient ways of doing it.

My PR was an attempt to add the optimized functionality directly into Dijkstra, but given the objections there it's now in GraphCentrality. Feel free to examine/use it there. Betweenness is by its nature expensive, but I think the approach I've taken in GraphCentrality is at least 50% better than the other implementations out there.